include/boost/geometry/algorithms/detail/overlay/handle_colocations.hpp

```diff
@@ -98,6 +97,7 @@ inline void cleanup_clusters(Turns& turns, Clusters& clusters)
     }
 
     remove_clusters(turns, clusters);
+    colocate_clusters(clusters, turns);
```

Comment:
This now looks a bit weird (by the order change).
I will change it in a later PR but I didn't want to do it in this one, to keep the size smaller.